django/db/models/fields/__init__.py

```diff
@@ -2019,10 +2026,11 @@ def rel_db_type(self, connection):
         if connection.features.related_fields_match_type:
             return self.db_type(connection)
         else:
-            return IntegerField().db_type(connection=connection)
+            integer_class = self.__class__.__bases__[1]
+            return integer_class().db_type(connection=connection)
```

Comment:
That seems a bit brittle wrt to `PositiveBigIntegerField` subclassing or subclasses.

What about using `__init_subclass__` to determine it from the MRO at class creation

```python
class PositiveIntegerRelDbTypeMixin:
    def __init_subclass__(cls, **kwargs):
        super().__init_subclass__(**kwargs)
        if getattr(cls, 'integer_field_class', None) is None:
            integer_field_class = next((
                parent for parent in cls.__mro__ if issubclass(parent, IntegerField)
            ), None)
```

and then

```suggestion
            return self.integer_field_class().db_type(connection=connection)
```